src/UseCases/CancelMembershipApplication/CancelMembershipApplicationUseCase.php

```diff
@@ -27,7 +31,63 @@ public function __construct( MembershipApplicationAuthorizer $authorizer,
 	}
 
 	public function cancelApplication( CancellationRequest $request ): CancellationResponse {
-		// TODO
+		$application = $this->getApplicationById( $request->getApplicationId() );
+
+		if ( $application === null ) {
+			return $this->newFailureResponse( $request );
+		}
+
+		$application->cancel();
+
+		try {
+			$this->repository->storeApplication( $application );
+		}
+		catch ( StoreMembershipApplicationException $ex ) {
+			// TODO: log?
+			return $this->newFailureResponse( $request );
+		}
+
+		$this->sendConfirmationEmail( $application );
```

Comment:
I still think that errors from the mailer should be caught here cause a message on the success page. Something like "We have canceled your application but could not send an email." Otherwise, the user would see an error page (exceptions will be caught by general error handling) and think that the cancellation failed, when in fact it succeeded.